app/HutchAgent/Services/WorkflowTriggerService.cs

```diff
@@ -1,265 +1,55 @@
-    var downloadAction = new ContextEntity(crate, downloadActionId);
-    crate.Add(downloadAction);
-    downloadAction.SetProperty("@type", "DownloadAction");
-    downloadAction.SetProperty("name", "Downloaded workflow RO-Crate via proxy");
-    downloadAction.SetProperty("startTime", downloadStartTime);
-
-    downloadAction.SetProperty("object", new Part()
-    {
-      Id = downloadAddress
-    });
-
-    downloadAction.SetProperty("agent", new Part()
-    {
-      Id = "http://proxy.example.com/"
-    });
-    // Set DownloadAction status to Active
-    _crateService.UpdateCrateActionStatus(ActionStatus.ActiveActionStatus, downloadAction);
-    using (var client = new HttpClient())
-    {
-      var clientStream = await client.GetStreamAsync(downloadAddress);
-      await using var file = File.OpenWrite(Path.Combine(cratePath, "workflows.zip"));
-      await clientStream.CopyToAsync(file);
-      _logger.LogInformation("Successfully downloaded workflow from Workflow Hub.");
-    }
-
-    //Set DownloadAction status to Completed
-    _crateService.UpdateCrateActionStatus(ActionStatus.CompletedActionStatus, downloadAction);
-
-    downloadAction.SetProperty("endTime", DateTime.Now);
-    downloadAction.SetProperty("result", new Part()
-    {
-      Id = Path.Combine("workflow", workflowId)
-    });
-    using (var archive = new ZipArchive(File.OpenRead(Path.Combine(cratePath, "workflows.zip"))))
-    {
-      Directory.CreateDirectory(Path.Combine(cratePath, "workflow", workflowId));
-      archive.ExtractToDirectory(Path.Combine(cratePath, "workflow", workflowId));
-      _logger.LogInformation($"Unpacked workflow to {Path.Combine(cratePath, "workflow", workflowId)}");
-    }
-
-    var workflowEntity = new Entity(crate);
-    workflowEntity.SetProperty("@id", Path.Combine("workflow", workflowId));
-    var property = crate.Entities[mainEntity.Id];
-    workflowEntity.SetProperty("sameAs", new Part()
-    {
-      Id = property.Id
-    });
-    workflowEntity.SetProperty("@type", property.Properties["@type"]);
-    workflowEntity.SetProperty("name", property.Properties["name"]);
-    workflowEntity.SetProperty("conformsTo", property.Properties["conformsTo"]);
-    workflowEntity.SetProperty("distribution", property.Properties["distribution"]);
-    crate.Add(workflowEntity);
-    crate.RootDataset.AppendTo("mentions", downloadAction);
-    crate.Save(cratePath);
-    _logger.LogInformation($"Saved updated RO-Crate to {cratePath}.");
-
-    await _WorkflowJobService.Create(WorkflowJob.Id, WorkflowJob.WorkingDirectory);
-
-    return (WorkflowJob, crate);
-  }
-
-  /// <summary>
-  /// Install and run WfExS given 
-  /// </summary>
-  /// <param name="stream"></param>
+  /// <param name="job"></param>
+  /// <param name="roCrate"></param>
   /// <exception cref="Exception"></exception>
-  public async Task TriggerWfexs(Stream stream)
+  public async Task TriggerWfexs(Data.Entities.WorkflowJob job,ROCrate roCrate)
```

Comment:
```suggestion
  public async Task TriggerWfexs(Models.WorkflowJob job,ROCrate roCrate)
```
Use the model rather than the direct representation of the db entity.

Reply:
> Use the model rather than the direct representation of the db entity.

.Get() method returns an entity when I call TriggerWfexs() so I pass it along

Reply:
I changed it on my PR. So I think if you change it here it should work out when we come t merge it all together.

Reply:
Everyone loves merge order dependencies